hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/commit/AbstractWriteHelper.java

```diff
@@ -81,10 +82,10 @@ public I combineOnCondition(
    * @return Collection of HoodieRecord already be deduplicated
    */
   public I deduplicateRecords(
-      I records, HoodieTable<T, I, K, O> table, int parallelism) {
-    return deduplicateRecords(records, table.getIndex(), parallelism);
+      I records, HoodieTable<T, I, K, O> table, int parallelism, HoodieWriteConfig config) {
```

Comment:
Is there a way to just pass the schema and not the whole write config around?

Reply:
the problem is Shema is not serializable,  one alternative is just pass the schema string